File: cpp/hot100/29.删除链表的倒数第 N 个结点.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode *dummy=new ListNode(0);
        dummy->next=head;
        ListNode *pre=dummy;
        ListNode *lat=dummy;
        for(int i=0;i<n;i++){
            if(!lat->next) return head;
            lat=lat->next;
        }
        while(lat->next){
            lat=lat->next;
            pre=pre->next;
        }
        pre->next=pre->next->next;
        return dummy->next;
    }
};

File: python/hot100/29.删除链表的倒数第 N 个结点.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy=ListNode(0,head)
        pre=dummy
        lat=dummy
        for i in range(n):
            lat=lat.next
        while lat.next:
            pre=pre.next
            lat=lat.next
        pre.next=pre.next.next
        return dummy.next